akka-stream/src/main/scala/akka/stream/impl/Sinks.scala

```diff
@@ -247,3 +224,78 @@ private[akka] final class HeadOptionStage[T] extends SinkStage[T, Future[Option[
     }, p.future)
   }
 }
+
+/**
+ * INTERNAL API
+ */
+private[akka] class QueueSink[T]() extends SinkStage[T, SinkQueue[T]]("queueSink") {
```

Comment:
This Sink is currently not reusable (i.e. if it is included in a graph that is included twice in a larger graph then things will break) due to state being kept inside the GraphStage—all state must be in the GraphStageLogic instead.